### DISCUSSION TITLE
Using USART1 for IrDA

### DISCUSSION THREAD

**Discussion by pablo-1424** [danak6jq/Seeed-LoRa-E5#20] [Q&A]
Hello,

I want to change the USART1 to IrDA mode. To do that I have change the Platform settings of the LoRaWAN to USART2 and that is all good. But when I generate the code and compile, it gives me an error in the usart_if.c file : 

C:/work/Seeed-LoRa-E5/Debug/../Core/Src/usart_if.c:217: undefined reference to `hdma_usart2_tx'

Do you have any idea to resolve this issue ?

Thanks for your code, that is my only problem,

Pablo

**Comment by danak6jq** [comment]
@pablo-1424 what changes exactly did you make to USART2? Did you add DMA support?

**Comment by danak6jq** [comment]
This is doubtlessly an issue with CubeMX code generation. It looks like USART2 is configured to use DMA for TX, but the DMA service function is not present. This makes me think that, after configuring USART2 and generating code from the IOC, some source file(s) was over-written with an older version (or mis-merged). In particular I'd look at Core/Src/usart.c - this is where CubeMX generates the DMA handler. 

This isn't an issue in the LoRaWAN project, it's a hiccup in source changes in your workspace. After running CubeMX and generating source, you should have hdma_usart2_tx in usart.c - is this not the case?
